### ISSUE TITLE
[R-package] WARNING on M1Mac CRAN job

### ISSUE THREAD

**Issue by jameslamb** [microsoft/LightGBM#5557]
## Description

The checks on the `M1Mac` CRAN flavor for `{lightgbm}` 3.3.3 shows one warning.

> checking whether package ‘lightgbm’ can be installed ... [94s/89s] WARNING
> Found the following significant warnings:
  ./include/LightGBM/utils/log.h:117:5: warning: 'vsprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use vsnprintf(3) instead. [-Wdeprecated-declarations]
See ‘/Users/ripley/R/packages/tests-devel/lightgbm.Rcheck/00install.out’ for details.

<img width="675" alt="Screen Shot 2022-10-26 at 12 59 23 PM" src="https://user-images.githubusercontent.com/7608904/198101470-6bb202e4-a71d-402d-b503-12c3875700fe.png">

## Reproducible example

* log link: https://www.stats.ox.ac.uk/pub/bdr/M1mac/lightgbm.out
* all statuses: https://cran.r-project.org/web/checks/check_results_lightgbm.html
* `M1Mac` job setup: https://www.stats.ox.ac.uk/pub/bdr/M1mac/README.txt

## Environment info

LightGBM version or commit hash: v3.3.3

## Additional Comments

LightGBM is not tested on M1 Macs in its CI today.

* #5035
* #5269
* #5328
* #5549

**Comment by jameslamb** [comment]
@shiyu1994 please check your email. Did you receive any emails from CRAN saying that we have to fix this?

**Comment by jameslamb** [comment]
I see the following at https://www.stats.ox.ac.uk/pub/bdr/M1mac/README.txt

> Note that Apple has deprecated C functions `sprintf` and `vsprintf` in
macOS 13 SDK: this also affect users of C++.  (And the much less
commonly used and widely deprecated `gets`, `mktemp` and `tmpnam`.)

**Comment by shiyu1994** [comment]
@jameslamb The last email that I received from CRAN is

```bash
Dear maintainer,
 
package lightgbm_3.3.3.tar.gz has been auto-processed and is pending an automated reverse dependency check. This service will typically respond to you within the next day. For technical reasons you may receive a second copy of this message when a team member triggers a new check.
 
Log dir: <https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwin-builder.r-project.org%2Fincoming_pretest%2Flightgbm_3.3.3_20221010_175239%2F&amp;data=05%7C01%7Cyushi2%40microsoft.com%7C1384c0da9d4a41f150cd08daaad9b0a9%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638010149253131546%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&amp;sdata=3QiP8ktZ34sppxB%2B5zn0yFQJ0JeduqQG5TXzvQ%2Fb%2Fpo%3D&amp;reserved=0>
The files will be removed after roughly 7 days.
Installation time in seconds: 203
Check time in seconds: 134 
R Under development (unstable) (2022-10-08 r83047 ucrt)
 
Pretests results:
Windows: <https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwin-builder.r-project.org%2Fincoming_pretest%2Flightgbm_3.3.3_20221010_175239%2FWindows%2F00check.log&amp;data=05%7C01%7Cyushi2%40microsoft.com%7C1384c0da9d4a41f150cd08daaad9b0a9%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638010149253131546%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&amp;sdata=BWN5Zh6kr1AOJj5rtkcRwXPq575%2FDGnvxWgjko4tKXQ%3D&amp;reserved=0>
Status: OK
Debian: <https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwin-builder.r-project.org%2Fincoming_pretest%2Flightgbm_3.3.3_20221010_175239%2FDebian%2F00check.log&amp;data=05%7C01%7Cyushi2%40microsoft.com%7C1384c0da9d4a41f150cd08daaad9b0a9%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638010149253131546%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&amp;sdata=9YxSStP2EtLlwTEBKakXlMDnXK6NjdXgclyFtZCusr4%3D&amp;reserved=0>
Status: OK
 
Last released version's CRAN status: OK: 6, NOTE: 6, ERROR: 1
See: <https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fcran.r-project.org%2Fweb%2Fchecks%2Fcheck_results_lightgbm.html&amp;data=05%7C01%7Cyushi2%40microsoft.com%7C1384c0da9d4a41f150cd08daaad9b0a9%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638010149253131546%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&amp;sdata=6r7z3uwj30H%2FeR3jEL470BILbldd668%2BAlhIH0WyNcU%3D&amp;reserved=0>
 
CRAN Web: <https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fcran.r-project.org%2Fpackage%3Dlightgbm&amp;data=05%7C01%7Cyushi2%40microsoft.com%7C1384c0da9d4a41f150cd08daaad9b0a9%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638010149253131546%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&amp;sdata=sLl93vfulvVyxMSlLb0r3R6XCVe7Kr7i4ptI%2FoXpWvo%3D&amp;reserved=0>
 
*** Strong rev. depends ***: cbl
 
Best regards,
CRAN teams' auto-check service

```

which was sent on Oct 11. I think the email did not indicate any error of our package.

**Comment by jameslamb** [comment]
Interesting, ok.

Thanks @shiyu1994 ! I guess the CRAN package is safe for now!

So this can be closed once #5561 is merged.

**Comment by jameslamb** [comment]
CRAN eventually did make us fix this. See #5618.

v3.3.4, now on CRAN, includes a fix and is passing all CRAN checks. See the links in the description of #5646 for more details.
